packages/rector-generator/src/Command/GenerateCommand.php

```diff
@@ -62,37 +72,63 @@ final class GenerateCommand extends Command
      */
     private $rectorRecipeProvider;
 
+    /**
+     * @var PackageNamesProvider
+     */
+    private $packageNamesProvider;
+
+    /**
+     * @var NodeTypesProvider
+     */
+    private $nodeTypesProvider;
+
+    /**
+     * @var SetsListProvider
+     */
+    private $setsListProvider;
+
     public function __construct(
         ComposerPackageAutoloadUpdater $composerPackageAutoloadUpdater,
         ConfigFilesystem $configFilesystem,
         FileGenerator $fileGenerator,
         OverrideGuard $overrideGuard,
-        SymfonyStyle $symfonyStyle,
         TemplateFinder $templateFinder,
         TemplateVariablesFactory $templateVariablesFactory,
-        RectorRecipeProvider $rectorRecipeProvider
+        RectorRecipeProvider $rectorRecipeProvider,
+        PackageNamesProvider $packageNamesProvider,
+        NodeTypesProvider $nodeTypesProvider,
+        SetsListProvider $setsListProvider
     ) {
         parent::__construct();
 
-        $this->symfonyStyle = $symfonyStyle;
         $this->templateVariablesFactory = $templateVariablesFactory;
         $this->composerPackageAutoloadUpdater = $composerPackageAutoloadUpdater;
         $this->templateFinder = $templateFinder;
         $this->configFilesystem = $configFilesystem;
         $this->overrideGuard = $overrideGuard;
         $this->fileGenerator = $fileGenerator;
         $this->rectorRecipeProvider = $rectorRecipeProvider;
+        $this->packageNamesProvider = $packageNamesProvider;
+        $this->nodeTypesProvider = $nodeTypesProvider;
+        $this->setsListProvider = $setsListProvider;
     }
 
     protected function configure(): void
     {
         $this->setAliases(['c', 'create', 'g']);
         $this->setDescription('[DEV] Create a new Rector, in a proper location, with new tests');
+        $this->addOption(
+            self::INTERACTIVE_MODE_NAME,
+            'i',
+            InputOption::VALUE_NONE,
+            'Turns on Interactive Mode - Rector will be generated based on responses to questions instead of using rector-recipe.php',
+        );
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $rectorRecipe = $this->rectorRecipeProvider->provide();
+        $this->symfonyStyle = new SymfonyStyle($input, $output);
```

Comment:
When `SymfonyStyle` was passed as service I had issues with testing the command using `CommandTester`. All inputs that I passed in the test using `$this->commandTester->setInputs()` where NOT passed to the same object and tests where not run as it was waiting for User Input.

Reply:
We should not create services manually in our code just for sake of testing. 
I don't use `CommandTester`, probably because of this design. 

Why do we need to test console responses (basically symfony/console)?

IMO testing the output is good enough.

Reply:
I've got lost then. That's how I understood [your comment](https://github.com/rectorphp/rector/pull/4931#issuecomment-749089779) from a few days ago. I don't know how test the final output of interactive command without passing input user data. I've looked through some other Symfony projects (like Maker Bundle) and most of them create SymfonyStyle instead of providing it in Dependency Injection.

Reply:
@TomasVotruba I've brought back `SymfonyStyle` as service but right now I am not sure how and what you want to test here , sorry. It uses the same Generator as normal Generate command, only asks questions for arguments using `SymfonyStyle` instead of getting them from config file.

Reply:
Yea, in Symfony projects the `SymfonyStyle` is often used as a value object. That usually leads to mix of input/output in various instances. That's why it's better to have `SymfonyStyle` as a service. One reliable service fo all.

Don't worry about it, I'll look into it :)